### PR TITLE
Fix #282 use modern arithmetic expansion in shell

### LINKED ISSUE #282: Shell arithmetic expansion
Both bash and sh documentation (Linux) says that $[expression] is an old syntax. They prefeer $((expression)). bash manpage says:

"The old format $[expression] is deprecated and will be removed in upcoming versions of bash."

sh manpage shows only $((expression)) syntax, not $[expression].

### PR DESCRIPTION
**Pull request by simonnagl**
If you want to update this in your other projects:

```sh
for file in `find "." -name "*.sh"`
do
    perl -pi -e 's/\$\[([^\[\]]+|([^\[\]]*?\[[^\[\]]+?\][^\[\]]*?)+)\]/\$(($1))/' "$file"
done
```

### CODE REVIEW

File: tests/stress.sh [hunk 1]
```diff
@@ -50,22 +50,22 @@ do
...
 
-		points=$[RANDOM * 2000 / 32767 + 2]
-		st=$[first + st]
-		et=$[first + et]
+		points=$((RANDOM * 2000 / 32767 + 2))
+		st=$((first + st))
+		et=$((first + et))
 
-		mode=$[RANDOM * ${#modes[@]} / 32767]
+		mode=$((RANDOM * ${#modes[@))} / 32767]
```
Comment (ktsaou) [inline tests/stress.sh]: hm... this is not right!

File: tests/stress.sh [hunk 2]
```diff
@@ -50,22 +50,22 @@ do
...
 		mode="${modes[$mode]}"
 
-		chart=$[RANDOM * ${#charts[@]} / 32767]
+		chart=$((RANDOM * ${#charts[@))} / 32767]
```
Comment (ktsaou) [inline tests/stress.sh]: again here

File: tests/stress.sh [hunk 3]
```diff
@@ -50,22 +50,22 @@ do
...
 		chart="${charts[$chart]}"
 
-		format=$[RANDOM * ${#formats[@]} / 32767]
+		format=$((RANDOM * ${#formats[@))} / 32767]
```
Comment (ktsaou) [inline tests/stress.sh]: and here

File: charts.d/apache.chart.sh [hunk 4]
```diff
@@ -47,21 +47,21 @@ apache_detect() {
...
-			'BytesPerReq')			apache_key_bytesperreq=$[i + 1] ;;
-			'BusyWorkers')			apache_key_busyworkers=$[i + 1] ;;
-			'IdleWorkers')			apache_key_idleworkers=$[i + 1];;
-			'ConnsTotal')			apache_key_connstotal=$[i + 1] ;;
-			'ConnsAsyncWriting')	apache_key_connsasyncwriting=$[i + 1] ;;
-			'ConnsAsyncKeepAlive')	apache_key_connsasynckeepalive=$[i + 1] ;;
-			'ConnsAsyncClosing')	apache_key_connsasyncclosing=$[i + 1] ;;
-			'Scoreboard')			apache_key_scoreboard=$[i] ;;
+			'Total Accesses') 		apache_key_accesses=$((i + 1)) ;;
+			'Total kBytes') 		apache_key_kbytes=$((i + 1)) ;;
```
Comment (ktsaou) [inline charts.d/apache.chart.sh]: this seems to be already in the project like that. probably a web view issue...

**Comment by ktsaou** [comment]
nice!
Please fix the broken ones and I'll merge it.

**Comment by simonnagl** [comment]
Done.
I also updated the regex from above. It now can handle arrays. If there are more braces in braces it won't work. Then one should use recursion. I do not know if this is possible on command-line perl without getting verry ugly.

**Comment by ktsaou** [comment]
There are a few more scripts in the `plugins.d` directory.
Can you do these do?

**Comment by ktsaou** [comment]
The big one should be `charts.d.plugin`.

**Comment by simonnagl** [comment]
Done. This was the last shell script with `$[...]` in the repo.

**Comment by ktsaou** [comment]
Nice! You are the best.
Have you added the file `plugins.d/skript` intentionally? Do we need it?

**Comment by simonnagl** [comment]
No, we do not need it. Just the script pasted above. Sorry for this error!

**Comment by ktsaou** [comment]
merged!
Thank you!

**Pull request merged**